### PR TITLE
Allow creating an object with a Cocina model

### PR DESCRIPTION
**Pull request by jcoyne**
## Why was this change made?

So that we can support creating with more values.

## Was the API documentation (openapi.yml) updated?
yes

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -1,3 +1,11 @@
+PATH
+  remote: ../cocina-models
```
Comment (mjgiarlo) [inline Gemfile.lock]: Does this file need to be updated? This looks like cruft from earlier when you might have been pointing at a local path?

File: app/services/cocina/object_creator.rb [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    def create(params)
+      obj = case params[:type]
+            when *Cocina::Models::DRO::TYPES
+              build_dro(params)
+            when *Cocina::Models::Collection::TYPES
+              build_collection(params)
+            when *Cocina::Models::AdminPolicy::TYPES
+              build_apo(params)
+            else
+              raise "Unknown type #{params[:type]}"
```
Comment (mjgiarlo) [inline app/services/cocina/object_creator.rb]: It looks like adding coverage for this line would make the build happier.

File: app/services/cocina/object_creator.rb [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+            end
+      if validate(obj)
+        af_model = create_from_model(obj)
+
+        # This will rebuild the cocina model from fedora, which shows we are only returning persisted data
+        return Mapper.build(af_model)
+      end
+
+      raise 'not valid'
```
Comment (mjgiarlo) [inline app/services/cocina/object_creator.rb]: It looks like adding coverage for this line would make the build happier.

File: app/services/cocina/object_creator.rb [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+
+    def create_from_model(obj)
+      af_object = if obj.admin_policy?
+                    create_apo(obj)
+                  elsif obj.dro?
+                    create_dro(obj)
+                  elsif obj.collection?
+                    create_collection(obj)
+                  else
+                    raise "unsupported type #{obj.type}"
```
Comment (mjgiarlo) [inline app/services/cocina/object_creator.rb]: It looks like adding coverage for this line would make the build happier.

File: app/services/cocina/object_creator.rb [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+    end
+
+    def catkey_for(obj)
+      obj.identification.catalogLinks&.find { |l| l.catalog == 'symphony' }&.catalogRecordId
+    end
+
+    def content_type_tag(type)
+      tag = case type
+            when Cocina::Models::Vocab.image
+              'Image'
```
Comment (mjgiarlo) [inline app/services/cocina/object_creator.rb]: It looks like only this branch is tested.

File: openapi.yml [hunk 6]
```diff
@@ -700,50 +700,11 @@ paths:
...
-                  type: array
-                  items:
-                    type: string
-                    enum:
-                      - descMetadata
-                  description: If 'descMetadata' is provided, then record is seeded from symphony
-                  example: ['descMetadata']
-              required:
-                - admin_policy
+              oneOf:
```
Comment (mjgiarlo) [inline openapi.yml]: Nice change!

File: openapi.yml [hunk 7]
```diff
@@ -871,6 +832,8 @@ components:
         - administrative
         - identification
         - structural
+    Description:
+      type: object
```
Comment (mjgiarlo) [inline openapi.yml]: Does this object require any structure or have any required props?

File: app/services/cocina/object_creator.rb (hunk 5 above)
Comment (jcoyne) [reply app/services/cocina/object_creator.rb]: @mjgiarlo  Is exhaustive code coverage required? I don't think those would be useful tests.

File: openapi.yml (hunk 7 above)
Comment (jcoyne) [reply openapi.yml]: Not until @arcadiafalcone tells us what it's going to look like.

File: app/services/cocina/object_creator.rb (hunk 5 above)
Comment (mjgiarlo) [reply app/services/cocina/object_creator.rb]: No, not required. These branches are all simple enough, and in fact... testing these exhaustively will increase the drag when refactoring these later (if we want to replace the conditionals with polymorphism, for example).

**Pull request merged**